File: comments/src/main/java/de/thm/arsnova/service/comment/service/RoomAccessService.java
package de.thm.arsnova.service.comment.service;

import de.thm.arsnova.service.comment.model.RoomAccess;
import de.thm.arsnova.service.comment.model.RoomAccessPK;
import de.thm.arsnova.service.comment.service.persistence.RoomAccessRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class RoomAccessService {
    private static final Logger logger = LoggerFactory.getLogger(RoomAccessService.class);

    final RoomAccessRepository repository;

    @Autowired
    public RoomAccessService(RoomAccessRepository repository) {
        this.repository = repository;
    }

    public RoomAccess get(RoomAccessPK pk) {
        return repository.findById(pk).orElse(new RoomAccess());
    }

    public RoomAccess create(RoomAccess roomAccess) {
        logger.trace("Creating RoomAccess: " + roomAccess.toString());
        return repository.save(roomAccess);
    }
}

File: websocket/src/main/kotlin/de/thm/arsnova/service/wsgateway/service/RoomAccessService.kt
package de.thm.arsnova.service.wsgateway.service

import de.thm.arsnova.service.wsgateway.config.WebSocketProperties
import de.thm.arsnova.service.wsgateway.model.RoomAccess
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate

@Service
class RoomAccessService(
		private val webSocketProperties: WebSocketProperties,
		private val restTemplate: RestTemplate
) {
	companion object {
		val roomAccessString = "roomaccess"
	}

	private val logger = LoggerFactory.getLogger(RoomAccessService::class.java)

	private var roomAccessGetEndpoint = "${webSocketProperties.httpClient.authService}/$roomAccessString"

	fun getRoomAccess(roomId: String, userId: String): RoomAccess? {
		val url = "$roomAccessGetEndpoint/$roomId/$userId"
		logger.trace("Querying auth service for room access with url: {}", url)
		return restTemplate.getForObject(url, RoomAccess::class.java)
	}
}
